test/procedure.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const metatests = require('metatests');
+const { Procedure } = require('../lib/procedure');
+
+metatests.testAsync('lib/procedure validate', async (test) => {
+  const script = () => ({
+    validate: ({ a, b }) => {
+      if (a % 3 === 0) throw new Error('Expected `a` to be multiple of 3');
+      if (b % 5 === 0) throw new Error('Expected `b` to be multiple of 5');
+    },
+
+    method: async ({ a, b }) => {
+      const result = a + b;
+      return result;
+    },
+  });
+  const procedure = new Procedure(script, require('../lib/application'));
```

Comment:
```suggestion

  const application = {
    Error,
    server: {
      semaphore: {
        await enter() {}
        leave() {}
      }
    }
  };
  const procedure = new Procedure(script, application);
```
We need just application stub.